lib/exit.js

```diff
@@ -26,12 +26,15 @@ export default class Exit {
   /**
    * params:
    * utxoId: string with 0x, e.g. 0x0000000000000000000000000000000000b447b980aea5accb5fd68789f6b099
-   * amount: plain JS number
+   * amount: BigInt | number
    * returns:
    * 64 byte buffer
    */
   static sigHashBuff(utxoId, amount) {
-    const valueBuf = util.setLengthLeft(util.toBuffer(amount), 32);
+    const valueBuf = util.setLengthLeft(
+      Buffer.from(util.padToEven(amount.toString(16)), 'hex'),
```

Comment:
would the buffer care if it is not even? length is fixed at 32 anyways, no?

Reply:
tests will fail without padding. 1000 → 0x3e8 → Buffer with 0x00...3e → wrong

don't even have time to think about